apps/bridge-dapp/src/containers/DepositContainer/DepositContainer.tsx

```diff
@@ -31,558 +31,541 @@ import { DepositContainerProps } from './types';
+          }
+        }}
+        onClose={() => setMainComponent(undefined)}
+      />
+    );
+  }, [chains, selectedSourceChain, setMainComponent, sourceChains]);
 
-      return 'Connect wallet';
-    }, [hasNoteAccount, isWalletConnected, isWrapFlow]);
+  const onMaxBtnClick = useCallback(() => {
+    setAmount(selectedTokenBalance ?? 0);
+  }, [selectedTokenBalance]);
 
-    const bridgingTokenProps = useMemo<
-      DepositCardProps['bridgingTokenProps']
-    >(() => {
-      if (!wrappableCurrency || !brideGovernedCurrency) {
-        return undefined;
-      }
-      const targetSymbol = brideGovernedCurrency.currency.view.symbol;
-
-      const tokens = getPossibleGovernedCurrencies(wrappableCurrency.id).map(
-        (currency): AssetType => ({
-          name: currency.view.name,
-          balance: balances[currency.id] ?? 0,
+  // Main action on click
+  const actionOnClick = useCallback(async () => {
+    // No wallet connected
+    if (!isWalletConnected) {
+      const sourceChains: ChainType[] = Object.values(chains).map((val) => {
+        return {
+          name: val.name,
+          symbol: currenciesConfig[val.nativeCurrencyId].symbol,
+        };
+      });
 
-          symbol: currency.view.symbol,
-        })
+      setMainComponent(<ChainSelectionWrapper sourceChains={sourceChains} />);
+      return;
+    }
+
+    // No note account exists
+    if (!hasNoteAccount) {
+      setNoteAccountModalOpen(true);
+      return;
+    }
+
+    if (
+      sourceChain &&
+      destChain &&
+      selectedToken &&
+      amount !== 0 &&
+      activeApi?.state?.activeBridge &&
+      activeChain
+    ) {
+      setIsGeneratingNote(true);
+      const chainId = calculateTypedChainId(
+        activeChain.chainType,
+        activeChain.chainId
+      );
+      const wrappbleTokenAddress =
+        wrappableCurrency?.getAddress(chainId) ?? undefined;
+      const newDepositPayload = await generateNote(
+        activeApi.state.activeBridge.targets[
+          calculateTypedChainId(sourceChain.chainType, sourceChain.chainId)
+        ],
+        calculateTypedChainId(destChain.chainType, destChain.chainId),
+        amount,
+        wrappbleTokenAddress
+      );
+      setIsGeneratingNote(false);
+      setMainComponent(
+        <DepositConfirmContainer
+          wrappingFlow={Boolean(wrappbleTokenAddress)}
+          wrappableTokenSymbol={governedCurrency?.view.symbol}
+          amount={amount}
+          token={selectedToken}
+          sourceChain={selectedSourceChain}
+          destChain={destChainInputValue}
+          depositPayload={newDepositPayload}
+        />
       );
+    }
+  }, [
+    isWalletConnected,
+    hasNoteAccount,
+    sourceChain,
+    destChain,
+    selectedToken,
+    amount,
+    activeApi?.state?.activeBridge,
+    activeChain,
+    chains,
+    setMainComponent,
+    setNoteAccountModalOpen,
+    generateNote,
+    selectedSourceChain,
+    destChainInputValue,
+    wrappableCurrency,
```

Comment:
React hook has a missing dependency: `'governedCurrency?.view.symbol'`. I think this can leak bugs.